Migration of hr_employee_exemption to version 8.0

Zwygart opened the pull request.

Commit:
[ADD] Module hr_employee_exemption

Commit:
Fix model name

Commit:
Fix security

Commit:
Fix method

Commit:
Link exemptions to rules

Commit:
Change logo

gurneyalex committed:
[FIX] fix Python SyntaxWarning

ghost commented:
:-1: You did not follow the procedure to migrate modules.
Refer to https://github.com/odoomrp/odoomrp-wip/wiki/Migration-to-OCA
Otherwise, you are overriding the git history/log of the module.

max3903 commented:
CLA is fixed

yvaucher commented:
@Zwygart Could you do a rebase instead of merge to keep the history clean. Here it's hard to see you diff

yvaucher commented:
@Zwygart This is migrated from where? I don't see this module in `__unported__` neither on 7.0 branch (I even had a look in 6.1)

max3903 commented:
@yvaucher This is migrated from the module in the 7.0 branch:
https://github.com/OCA/hr/tree/7.0/hr_employee_exemption

yvaucher commented:
@max3903 Thanks I missed it

And by the way some PEP8 to fix: https://travis-ci.org/OCA/hr/jobs/75007923#L164

Zwygart committed:
Bug fixing for travis errors.

Zwygart committed:
Added id to compute_rule

Zwygart commented:
And now? All the tests are passed. The migration is done. Anything else to do on this module?
If not : YAY, I contributed my first contribution to Odoo and OCA. :)

pedrobaeza commented:
Now people need to review and you need at least 2 approvals and 5 days to be merged. Please be patient. The time you wait here can be the time to detect possible glitches or problems.

yvaucher commented:
@Zwygart Could you do a `git rebase -i origin/8.0` then `git push -f` please? This would show use the correct diff and commits

ghost commented:
@Zwygart Thanks for your work.

Your pull request concerns 3 modules. Please, 1 module per pull request. Also, with your commit, we loose the git history. This is bad, because you override what Savoir-faire Linux has done on this module. Same thing for other modules you modify in this PR. As a respect to the original authors of the modules, use git filter-branch. Refer to https://github.com/odoomrp/odoomrp-wip/wiki/Migration-to-OCA. If you have issues with this procedure, ask and it will be my pleasure to help.

@yvaucher could you put back the needs fixing label ?

Zwygart commented:
Yeah, I definitely have issues with this procedure. Since I modified only one module. You say 3 are modified. I followed said procedure. With troubles mentioned earlier. I tried methods from the answers. Still don't seem to be the right way. 

It is a really small module, but seems very hard to submit it correctly. Good practice for when I'll try with others module. Still I want to success. 

So, sorry for using that much time and explication from you. At least each time I came further. So I got that going, which is nice. 

Thanks for your comprehension and clear help.

ghost commented:
@Zwygart Ok, here is what I would do.

First, create a new branch with filter branch:

git checkout origin/7.0
git checkout -b my_working_branch
git filter-branch --subdirectory-filter hr_employee_exemption -f
git filter-branch -f --tree-filter 'mkdir -v hr_employee_exemption ; git mv -k \* hr_employee_exemption' HEAD
git rebase 8.0

Then apply your changes to the module:

git checkout 8.0-hr_employee_exemption
cp -R hr_employee_exemption/ ~/backup/
git checkout my_working_branch
rm -Rf hr_employee_exemption
cp -R ~/backup/hr_employee_exemption hr_employee_exemption

Then commit and push:

git commit -am "Port hr_employee_exemption to 8.0"
git branch -D 8.0-hr_employee_exemption
git checkout -b 8.0-hr_employee_exemption
git push -f your_remote HEAD

Zwygart commented:
Ok, I executed all of your commands. It just failed at the first one, but all the other ones went smooth.

Zwygart commented:
Now, what's that supposed to mean? It executed perfectly. It still executes correctly on my machine.

yvaucher commented:
@Zwygart It seems you did a rebase on your local version of 8.0 branch

To fix it you need to rebase on remote 8.0 version (the OCA one on github)

```
# update object and refs
git fetch origin
# here you should have some output showing some changes
# then rebase your branch on top of the up to date 8.0
git rebase -i origin/8.0
```

You will encounter conflicts, read the instruction givent to you when doing `git status`

Zwygart committed:
Migration to version 8.0

Conflicts:
	hr_employee_exemption/models/hr_employee.py
	hr_employee_exemption/models/hr_employee_exemption.py
	hr_employee_exemption/models/hr_income_tax_exemption.py
	hr_employee_exemption/models/hr_salary_rule.py
	hr_employee_exemption/tests/__init__.py

Zwygart committed:
Corrections to respects the travis errors.

Conflicts:
	hr_employee_exemption/tests/__init__.py

Zwygart commented:
Ok, giving it another try.

dufresnedavid commented on hr_employee_exemption/models/hr_salary_rule.py:
```diff
@@ -0,0 +1,47 @@
+        """ Check whether the employee is exempted for the given rule
+        """
+
+        assert len(self) == 1, 'must be called with a single employee'
```
replace this whith self.ensure_one()

dufresnedavid commented on hr_employee_exemption/models/hr_employee.py:
```diff
@@ -0,0 +1,54 @@
+        """
+
+        assert len(self) == 1, 'must be called with a single employee'
+
```
replace this with self.ensure_one()

dufresnedavid commented on hr_employee_exemption/tests/__init__.py:
```diff
@@ -0,0 +1,28 @@
+
+checks = [
+    test_hr_employee_exemption,
+]
```
remove "checks = [test_hr_employee_exemption, ]". This is deprecated in v8.

dufresnedavid commented on hr_employee_exemption/tests/test_hr_employee_exemption.py:
```diff
@@ -0,0 +1,215 @@
+        self.contract_model = self.registry('hr.contract')
+        self.structure_model = self.registry("hr.payroll.structure")
+        self.payslip_model = self.registry('hr.payslip')
+        self.user_model = self.registry("res.users")
```
replace self.registry(...) with self.env[...] and remove cr, uid, context all over the file.

ghost commented:
also, the icon must be moved to ./static/descripton instead of ./static/src/img

dufresnedavid commented on hr_employee_exemption/i18n/fr.po:
```diff
@@ -0,0 +1,89 @@
+#. module: hr_employee_exemption
+#: field:hr.employee.exemption,date_to:0
+msgid "Date To"
+msgstr "Date de début"
```
I made this mistake in v7. I wrote Date de début instead of Date de fin. You may fix it.

dufresnedavid commented on hr_employee_exemption/i18n/fr.po:
```diff
@@ -0,0 +1,89 @@
+#. module: hr_employee_exemption
+#: field:hr.employee.exemption,date_from:0
+msgid "Date From"
+msgstr "Date de fin"
```
same thing here

ghost commented:
@Zwygart Thanks for your perseverance. Pointed a few details to fix.

Zwygart committed:
Correction to respect 8.0 in tests
Correction tu french translation

Zwygart committed:
Correct some errors : remove args on context_get

Zwygart committed:
Removed context on test

Zwygart committed:
Add id at tests

Zwygart committed:
Correct .id

Zwygart committed:
Add id to line 104

Zwygart committed:
context at payslip

yvaucher commented on hr_employee_exemption/__openerp__.py:
```diff
@@ -0,0 +1,47 @@
+    'category': 'Generic Modules/Human Resources',
+    'author': "Savoir-faire Linux, Odoo Community Association (OCA)",
+    'website': 'https://www.savoirfairelinux.com',
+    'description': """
```
Please remove description key from this dictionary. Description will be taken from README.rst

ghost commented:
@Zwygart are you still working on this ?

Zwygart commented:
To be honest : not really.

It's not easy and more time consuming than I tought. If someone else wants
to work on this : go ahead.

I will probably try later to give Odoo some help, but for now someone else
can continue the work.

Gideon Zwygart

2015-10-15 8:06 GMT-04:00 David Dufresne notifications@github.com:

> @Zwygart https://github.com/Zwygart are you still working on this ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/hr/pull/121#issuecomment-148367346.

## 

Zwygart

ghost commented:
@Zwygart I will take the lead on this PR. Thanks for your hard work.

ghost commented:
moved to https://github.com/OCA/hr/pull/165
